synapse/rest/client/v1/admin.py

```diff
@@ -730,6 +730,68 @@ def on_GET(self, request, target_user_id):
         defer.returnValue((200, ret))
 
 
+class ServerNoticeRestServlet(ClientV1RestServlet):
+    PATTERNS = client_path_patterns("/admin/send_server_notice(/(?P<user_id>[^/]+))?")
```

Comment:
I'm trying to figure out whether there is a difference in behaviour between `/admin/send_server_notice/` (with a trailing slash and an empty user_id) and `/admin/send_server_notice` (with no trailing slash). AFAICT the former will fail, but the docs say it is valid?

I'd suggest that you should not accept `/admin/send_server_notice/`, and the docs should be updated to clarify.